src/Lykke.Service.ClientSearch.Services/FullTextSearch/Indexer.cs

```diff
@@ -0,0 +1,170 @@
+        }
+
+        public void Initialize()
+        {
+            Task task = Task.Factory.StartNew(async () =>
+            {
+                await LoadAllPersonalDataForIndexing();
+            });
+        }
+
+        private void InitializeIndexFieldTypes()
+        {
+            storeFieldType = new FieldType();
+            storeFieldType.IndexOptions = IndexOptions.DOCS_ONLY;
+            storeFieldType.IsIndexed = true;
+            storeFieldType.IsStored = true;
+            storeFieldType.IsTokenized = false;
+            storeFieldType.OmitNorms = true;
+            storeFieldType.Freeze();
+
+            searchFieldType = new FieldType();
+            searchFieldType.IndexOptions = IndexOptions.DOCS_ONLY;
+            searchFieldType.IsIndexed = true;
+            searchFieldType.IsStored = true;
+            searchFieldType.IsTokenized = true;
+            searchFieldType.OmitNorms = false;
+            searchFieldType.Freeze();
+
+            phraseSearchFieldType = new FieldType();
+            phraseSearchFieldType.IndexOptions = IndexOptions.DOCS_AND_FREQS_AND_POSITIONS;
+            phraseSearchFieldType.IsIndexed = true;
+            phraseSearchFieldType.IsStored = true;
+            phraseSearchFieldType.IsTokenized = true;
+            phraseSearchFieldType.OmitNorms = true;
+            phraseSearchFieldType.Freeze();
+        }
+
+        private void CreateIndex(IEnumerable<IPersonalData> docsToIndex, IEnumerable<string> docsToDelete)
+        {
+            List<string> indexedFields = new List<string>();
+
+            using (var wAnalyzer = new StandardAnalyzer(LuceneVersion.LUCENE_48))
+            {
+                IndexWriterConfig config = new IndexWriterConfig(LuceneVersion.LUCENE_48, wAnalyzer);
+                config.OpenMode = OpenMode.CREATE_OR_APPEND;
+                config.Similarity = new DefaultSimilarity();
+
+                using (var writer = new IndexWriter(IndexDirectory, config))
+                {
+                    if (docsToIndex != null)
+                    {
+                        foreach (IPersonalData pd in docsToIndex)
+                        {
+                            bool somethingToIndex = false;
+
+                            var doc = new Document();
+                            doc.Add(new Field("ClientId", pd.Id, storeFieldType));
+
+                            string nameAndDoB = CreateNameAndDOBToIndex(pd);
+                            if (nameAndDoB != null)
+                            {
+                                indexedFields.Add(nameAndDoB);
+                                doc.Add(new Field("ClientNameAndDayOfBirth", nameAndDoB, phraseSearchFieldType));
+                                somethingToIndex = true;
+                            }
+
+                            if (somethingToIndex)
+                            {
+                                writer.UpdateDocument(new Term("ClientId", pd.Id), doc, wAnalyzer);

```

Comment:
The flag somethingToIndex is redundant. This line can be moved to pos 104 without result changing

Reply:
deleted